sequoia/http.py

```diff
@@ -57,7 +60,8 @@ def __init__(self, auth, session=None, proxies=None, user_agent=None, get_delay=
         if user_agent is not None:
             self.user_agent = user_agent + self.user_agent
 
-        self.backoff_strategy = backoff_strategy or HttpExecutor.DEFAULT_BACKOFF_CONF
+        self.backoff_strategy = copy.deepcopy(backoff_strategy) or HttpExecutor.DEFAULT_BACKOFF_CONF
+        self._remove_none_to_avoid_infinite_retries()
```

Comment:
I would put these two lines inside a method called `set_backoff_strategy`, what do you think?